flow/testing/mock_layer.h

```diff
@@ -57,6 +65,60 @@ class MockLayer : public Layer {
   FML_DISALLOW_COPY_AND_ASSIGN(MockLayer);
 };
 
+class MockCacheableContainerLayer : public CacheableContainerLayer {
+ public:
+  // if render more than 3 frames, try to cache itself.
+  // if less 3 frames, cache his children
+  explicit MockCacheableContainerLayer() : CacheableContainerLayer(3, true) {}
+
+  void Preroll(PrerollContext* context, const SkMatrix& matrix) override;
+};
+
+class MockCacheableContainerLayer2 : public CacheableContainerLayer {
```

Comment:
Needs either a better name or to collapse with the other MockCacheableContainer class.